### DISCUSSION TITLE
Nightly releases in main repo

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1535] [Ideas]
@kagikn I know that on June 12 you said that one of the reasons you don't release nightly versions to the main repo is that you don't want to spam with notifications users following the repo for releases, but on June 28 ([nightly89](https://github.com/scripthookvdotnet/scripthookvdotnet-nightly/releases/tag/v3.6.0-nightly.89)+) you fixed the issue of not being able to run any scripts in GTAV b3258 or later. Due to this, the vast majority of active and logged-in GitHub users who follow SHVDN for releases follow the nightly repo. I think that pushing releases to the main repo would, alongside compacting the whole project, make a nice version history if you choose to copy existing releases (although this would require renaming all ''.6.0'' nightly tags and releases to ''.7.0...'' to conform to Semantic Versioning). Keeping the nightly repo as well would keep [the ''latest nightly release'' link](https://github.com/scripthookvdotnet/scripthookvdotnet-nightly/releases/latest) still working.

**Comment by kagikn** [comment]
This would be inconvenient for users until v3.7.0 gets released, but I'm not going to push nightly releases to the main repo unless GitHub add support for receiving only releases that aren't mark as pre-releases. Once v3.7.0 gets released, they won't have to use a nightly version to avoid the crashing problem for the compatibility problem that [v3.6.0-nightly.89](https://github.com/scripthookvdotnet/scripthookvdotnet-nightly/releases/tag/v3.6.0-nightly.89) resolved.

There's at least one reason to distinguish nightly versions from stable ones, which is, that script devs should stick to stable versions when developing scripts unless they are testing some nightly APIs (of course there's no compatibility problems even if they run their scripts within a nightly SHVDN). I know there's script devs that build some their public scripts against nightly versions, but they shouldn't do that and no compatibility support will be provided for nightly-only APIs. I would be happy if script devs could propose their ideas to improve nightly-only APIs though, even if they have breaking changes.

**Reply by Michael21107** [reply]
I completely accept the decision and also despise devs writing scripts against nightly APIs.
